geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/iterators/AggregatingScan.scala

```diff
@@ -11,41 +11,40 @@ package org.locationtech.geomesa.index.iterators
 import java.io.Closeable
 
 import com.typesafe.scalalogging.LazyLogging
+import org.geotools.data.DataUtilities
 import org.geotools.filter.text.ecql.ECQL
 import org.locationtech.geomesa.features.SerializationOption.SerializationOptions
 import org.locationtech.geomesa.features.TransformSimpleFeature
 import org.locationtech.geomesa.features.kryo.KryoBufferSimpleFeature
 import org.locationtech.geomesa.index.api.GeoMesaFeatureIndex
-import org.locationtech.geomesa.index.iterators.AggregatingScan.RowValue
+import org.locationtech.geomesa.index.iterators.AggregatingScan.{AggregateCallback, CqlSampleValidator, CqlValidator, RowValidator, RowValue, SampleValidator, ValidateAll}
 import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
 import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
 import org.opengis.filter.Filter
 
+import scala.util.Try
 import scala.util.control.NonFatal
 
 trait AggregatingScan[T <: AggregatingScan.Result]
     extends SamplingIterator with ConfiguredScan with Closeable with LazyLogging {
 
   import AggregatingScan.Configuration._
 
-  private var sft: SimpleFeatureType = _
-  private var transformSft: SimpleFeatureType = _
-  // note: index won't have a hook to the data store, so some operations aren't available
-  private var index: GeoMesaFeatureIndex[_, _] = _
-
-  private var validate: SimpleFeature => Boolean = _
+  private var validate: RowValidator = _
 
   // our accumulated result
   private var result: T = _
 
+  private var batchSize: Int = _
+
   private var reusableSf: KryoBufferSimpleFeature = _
-  private var reusableTransformSf: TransformSimpleFeature = _
-  private var hasTransform: Boolean = _
+  private var aggregateSf: SimpleFeature = _
 
   override def init(options: Map[String, String]): Unit = {
     val spec = options(SftOpt)
-    sft = IteratorCache.sft(spec)
-    index = options.get(IndexSftOpt) match {
+    val sft = IteratorCache.sft(spec)
+    // note: index won't have a hook to the data store, so some operations aren't available
```

Comment:
Any particular operations worth noting?

Reply:
it doesn't affect the code here, just a warning for the future. it would be things like `getTablesNames` that need to read the catalog metadata